salesforcesoap/src/org/wso2/ballerina/connectors/salesforcesoap/salesforcesoapConnector.bal

```diff
@@ -0,0 +1,502 @@
+
+            map n = {"ns":"urn:partner.soap.sforce.com"};
+            sessionId = xmls:getString(loginResponse, "/ns:loginResponse/ns:result/ns:sessionId/text()", n);
+            session = `<urn:SessionHeader xmlns:urn="urn:partner.soap.sforce.com"><urn:sessionId>${sessionId}</urn:sessionId></urn:SessionHeader>`;
+            serverUrl = xmls:getString(loginResponse, "/ns:loginResponse/ns:result/ns:serverUrl/text()", n);
+        }
+        headers[headerCount] = session;
+        xml payload = `<urn:describeSObjects xmlns:urn="urn:partner.soap.sforce.com">
+                           <urn:sObjectType>${type}</urn:sObjectType>
+                       </urn:describeSObjects>`;
+        xml soapResponse = soap:ClientConnector.send(soapConnector, headers, payload, "''",
+        serverUrl, soapVersion);
+        if (isSessionExpired(soapResponse)) {
+            xml loginResponse = login(username, password, loginUrl, soapVersion);
+
+            map n = {"ns":"urn:partner.soap.sforce.com"};
+            sessionId = xmls:getString(loginResponse, "/ns:loginResponse/ns:result/ns:sessionId/text()", n);
+            session = `<urn:SessionHeader xmlns:urn="urn:partner.soap.sforce.com"><urn:sessionId>${sessionId}</urn:sessionId></urn:SessionHeader>`;
+            serverUrl = xmls:getString(loginResponse, "/ns:loginResponse/ns:result/ns:serverUrl/text()", n);
+            headers[headerCount] = session;
+            soapResponse = soap:ClientConnector.send(soapConnector, headers, payload, "''",
+                                                     serverUrl, soapVersion);
+        }
+
+         return soapResponse;
+    }
+
+    @doc:Description("Adds one or more new records to your organization’s data")
+    @doc:Param("s: The salesforcesoap connector instance")
+    @doc:Param("headers: Soap header values")
+    @doc:Param("type: Name of SObject")
+    @doc:Param("name: Name of record")
+    @doc:Return("response message")
+    action createRecord (ClientConnector s, xml[] headers, string type, string name) (xml) {
+
+        int headerCount = arrays:length(headers);
+        if (serverUrl == ""){
+            xml loginResponse = login(username, password, loginUrl, soapVersion);
+
+            map n = {"ns":"urn:partner.soap.sforce.com"};
+            sessionId = xmls:getString(loginResponse, "/ns:loginResponse/ns:result/ns:sessionId/text()", n);
+            session = `<urn:SessionHeader xmlns:urn="urn:partner.soap.sforce.com"><urn:sessionId>${sessionId}</urn:sessionId></urn:SessionHeader>`;
+            serverUrl = xmls:getString(loginResponse, "/ns:loginResponse/ns:result/ns:serverUrl/text()", n);
+        }
+        headers[headerCount] = session;
+        xml payload = `<urn:create xmlns:urn="urn:partner.soap.sforce.com" xmlns:urn1="urn:sobject.partner.soap.sforce.com">
+                           <urn:sObjects>
+                               <urn1:type>${type}</urn1:type>
+                               <urn1:Name>${name}</urn1:Name>
```

Comment:
As we discussed in the yesterday's meeting, we will be accepting mandatory parameters for an action as action parameters and the remainder as a struct. Once this approach is finalized, we can request a struct from the user with all the remaining parameters. For the today's release we will include only the mandatory parameters.